MSMAll/scripts/SingleSubjectConcat.sh

```diff
@@ -31,504 +31,171 @@
-						mGlobalPath="${HCPPIPEDIR}/global/matlab"
-						
-						matlabCode="addpath '$HCPCIFTIRWDIR'; addpath '$mGlobalPath'; addpath '$mPath';
-						ComputeVN('${cleandtseries}', '${bias}', '${ICAtcs}', '${ICANoise}', '${OutputVN}', '${Caret7_Command}');"
-						
-						log_Msg "$matlabCode"
-						"${interpreter[@]}" <<<"$matlabCode"
-						;;
-					
-					*)
-						# Unsupported MATLAB run mode
-						log_Err_Abort "Unsupported MATLAB run mode value: ${MatlabRunMode}"
-						;;
-				esac
-				
-			else
-				if [ ! -e ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_vn.dscalar.nii ] ; then
-					log_Err "Variance Normalization file doesn't exist:"
-					log_Err "${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_vn.dscalar.nii"
-					log_Err_Abort "You need to run RestingStateStats to generate it or set ComputeVarianceNormalization to YES"
-				fi
-				OutputVN="${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_vn.dscalar.nii"
-			fi
-			
-			MATHVN=" / max(VN,0.001)"
-			VarVN="-var VN ${OutputVN} -select 1 1 -repeat"
-		else
-			MATHVN=""
-			VarVN=""
-		fi
-      
-		MATH="((TCS${MATHDemean})${MATHRB})${MATHVN}"
-		log_Msg "MATH: ${MATH}"
-    
-		${Caret7_Command} -cifti-math "${MATH}" ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${OutputProcSTRING}.dtseries.nii -var TCS ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}.dtseries.nii ${VarDemean} ${VarRB} ${VarVN} 
-
-		if [ "${Demean}" = "YES" ] ; then
-			rm ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_mean.dscalar.nii
-		fi
-
-		if [ "${VarianceNormalization}" = "YES" ] ; then
-			if [ "${ComputeVarianceNormalization}" = "YES" ] ; then
-				rm ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_vn_tempcompute.dscalar.nii
-			fi
-		fi
-
-		MergeSTRING=`echo "${MergeSTRING} -cifti ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${OutputProcSTRING}.dtseries.nii"`
-		
-	done
-	mkdir -p "${OutputFolder}"
-	${Caret7_Command} -cifti-merge ${OutputFolder}/${OutputfMRIName}${fMRIProcSTRING}${OutputProcSTRING}.dtseries.nii ${MergeSTRING}
-}
-
-# ------------------------------------------------------------------------------
-#  "Global" processing - everything above here should be in a function
-# ------------------------------------------------------------------------------
-
-# Establish defaults
-G_DEFAULT_MATLAB_RUN_MODE=1		# Use interpreted MATLAB
-
-# Set global variables
-g_script_name=$(basename "${0}")
-
-# Allow script to return a Usage statement, before any other output
-if [ "$#" = "0" ]; then
-    show_usage
-    exit 1
+done
+if [[ "$EndFrame" == "" ]]; then
+	EndFrame="$minFrames"
+	log_Msg "EndFrame: ${EndFrame}"
 fi
-
-# Verify that HCPPIPEDIR environment variable is set
-if [ -z "${HCPPIPEDIR}" ]; then
-	echo "${g_script_name}: ABORTING: HCPPIPEDIR environment variable must be set"
-	exit 1
+# check valid frame range
+if [ $((StartFrame)) -gt "${minFrames}" ] || [ $((EndFrame)) -gt "${minFrames}" ] || [ $((StartFrame)) -lt 1 ] || [ $((EndFrame)) -lt 1 ] || [ $((StartFrame)) -gt $((EndFrame)) ]; then
```

Comment:
This would be cleaner like `if ((StartFrame > minFrames || EndFrame > minFrames`, etc.

However, this doesn't seem to allow the situation of concatenating full runs when the inputs aren't exactly the same length, and `--start-frame` and `--end-frame` aren't specified.  I think we should support this situation, which means preserving the original `--end-frame` default empty string, instead of overwriting that specific input variable with the computed minimum number of frames, and modifying the logic in the loop to distinguish these cases (and not run the `-cifti-merge` in the default case).